Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/shared/create-test-footer/create-test-footer.component.ts

```diff
@@ -1,8 +1,106 @@
-﻿import { Component } from '@angular/core';
+﻿import { Component, OnInit, ViewChild, Input } from '@angular/core';
+import { MdDialog, MdSnackBar } from '@angular/material';
+import { Test } from '../../tests.model';
+import { TestService } from '../../tests.service';
+import { ActivatedRoute } from '@angular/router';
+import { TestLaunchDialogComponent } from '../../test-settings/test-launch-dialog.component';
+import { Router } from '@angular/router';
+import { NgForm } from '@angular/forms';
+
 
 @Component({
     moduleId: module.id,
     selector: 'create-test-footer',
     templateUrl: 'create-test-footer.html',
 })
-export class CreateTestFooterComponent { }
+export class CreateTestFooterComponent implements OnInit {
+    testSettings: Test;
+    testId: number;
+    testSettingsUpdatedMessage: string;
+    isTestSection: boolean;
+    isTestQuestion: boolean;
+    isTestSettings: boolean;
+    @Input('settingsForm')
```

Comment:
on taking this form as input property, only the invalid angular 2 validations applied are passed from the test settings component to the create-test-footer component.